### PR TITLE
Add S3ObjectVersioning rule to CFRipper

### PR DESCRIPTION
**Pull request by ocrawford555**
Add S3ObjectVersioning rule to CFRipper. Tests updated.

### CODE REVIEW

**Review by carleslopez — DISMISSED** [review]

**Review by unlobito — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Changelog
 All notable changes to this project will be documented in this file.
 
+## [1.0.10] - 2021-09-XX
```
Comment (jsoucheiron) [inline CHANGELOG.md:4]: New feature, minor bump

File: cfripper/rules/s3_object_versioning.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    ENABLED_STATUS = "Enabled"
+    GRANULARITY = RuleGranularity.RESOURCE
+    REASON = "S3 Bucket {} is required to have object versioning enabled"
+    RISK_VALUE = RuleRisk.LOW
+    SUSPENDED_STATUS = "Suspended"
+
+    def invoke(self, cfmodel: CFModel, extras: Optional[Dict] = None) -> Result:
+        result = Result()
+        for logical_id, resource in cfmodel.resources_filtered_by_type(("AWS::S3::Bucket",)).items():
```
Comment (jsoucheiron) [inline cfripper/rules/s3_object_versioning.py:55]: Don't we have an abstract class that only applies a rule to Resources of a specific type?

**Review by jsoucheiron — CHANGES_REQUESTED** [review]

**Review by ignaciobolonio — APPROVED** [review]
LGTM

File: cfripper/rules/s3_object_versioning.py (hunk 2 above)
Comment (ocrawford555) [reply cfripper/rules/s3_object_versioning.py:55]: We do but we _currently_ don't have a `Resource` in `pycfmodel` for S3 Bucket (only S3 Bucket Policy). We would need to add that. Should that be done as part of this work?

Comment (jsoucheiron) [reply cfripper/rules/s3_object_versioning.py:55]: I'd say so, we had never had a reason to do it. Now we do

**Review by unlobito — APPROVED** [review]

**Review by carleslopez — APPROVED** [review]

**Review by jsoucheiron — APPROVED** [review]
Nice!

**Pull request merged**